### PR TITLE
Iet smichanges

### PR DESCRIPTION
**Pull request by manoj-freyr**
Changes to align rocm-smi and hip enumerations

### CODE REVIEW

File: iet.so/include/action.h [hunk 1]
```diff
@@ -118,8 +118,8 @@ class iet_action: public rvs::actionbase {
     //! list of GPUs (along with some identification data) which are
     //! selected for EDPp test
     std::vector<gpu_hwmon_info> edpp_gpus;
-
-
+    std::map<int, int> hip_to_smi_idxs;
```
Comment (ashutom) [inline iet.so/include/action.h:121]: I am not sure of the norms followed in RVS still,
for , hip_to_smi_idxs;
we can follow Hungarian notation of 'm'  to indicat its member of that class 
( never confuse type with kind in Hungarian : https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/)

File: iet.so/include/iet_worker.h [hunk 2]
```diff
@@ -62,7 +62,12 @@ class IETWorker : public rvs::ThreadBase {
     }
     //! returns the GPU index
     int get_gpu_device_index(void) { return gpu_device_index; }
-
+    //! sets the GPU smi index
+    void set_smi_device_index(int _smi_device_index) {
```
Comment (ashutom) [inline iet.so/include/iet_worker.h:66]: void set_smi_device_index(int _smi_device_index)   should always be

void set_smi_device_index(const int&  _smi_device_index)

File: iet.so/include/iet_worker.h [hunk 3]
```diff
@@ -208,14 +213,15 @@ class IETWorker : public rvs::ThreadBase {
...
-
  protected:
     std::unique_ptr<rvs_blas> gpu_blas;
 
     //! name of the action
     std::string action_name;
     //! index of the GPU (as reported by HIP API) that will run the EDPp test
     int gpu_device_index;
+    //! index of GPU (in view of smi lib) which is sometimes different to above index
+    int smi_device_index;
```
Comment (ashutom) [inline iet.so/include/iet_worker.h:224]: Same comment as before

File: iet.so/include/iet_worker.h (hunk 2 above)
Comment (manoj-freyr) [reply iet.so/include/iet_worker.h:66]: for simpler and native data types like int, char , I dont think pass by const ref will differ much from pass by value

File: iet.so/include/iet_worker.h (hunk 3 above)
Comment (manoj-freyr) [reply iet.so/include/iet_worker.h:224]: agreed, but kept naming convention consistent with the existing fields

**Review by ashutom — COMMENTED** [review]
Check individual comments

File: iet.so/include/action.h (hunk 1 above)
Comment (manoj-freyr) [reply iet.so/include/action.h:121]: agreed, but kept naming convention consistent with the existing fields

File: iet.so/include/iet_worker.h (hunk 2 above)
Comment (ashutom) [reply iet.so/include/iet_worker.h:66]: agreed,  but like 'winning is a habit' so is 'good coding ways' should also be :)
you may leave the code currently as it is

**Pull request merged**